### PR TITLE
184347112 Ft product expired date 

### LINKED ISSUE #52: The products should stop listing if the expiry date has passed
##  Why:
Expired Products should never appear on the list of the avilable products

## Description
Provided a product 
GIVEN. the expiration date has passed
   THEN:
       - the expired flag should be turned on
       - the product does not list on the available products list

Acceptance criteria.
Given the expiration date has passed, the product should be removed from 
the list of available products

Then:
- email is sent to the marchant

Dev notes:
- Use Crone jobs 
- Use Node events to implement the product lifecycle update"

### PR DESCRIPTION
**Pull request by UwaweTera**
### What does this PR do?

Expiration product

### Description of Task to be completed?

Expired Products should never appear on the list of the avilable products

### How should this be manually tested?

Provides steps on how this feature can tested.

### Any background context you want to provide?

Any other information you can give about the task.

### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/n/projects/2628461/stories/184347112

### Questions:

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code is formatted correctly
- [x] My changes doesn't have any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Ntarecedrick — DISMISSED** [review]

**Review by jimmymut — DISMISSED** [review]

**Review by NG-Solennel — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -62,6 +63,8 @@
     "morgan": "^1.10.0",
     "multer": "^1.4.5-lts.1",
     "multer-storage-cloudinary": "^4.0.0",
+    "node-cron": "^3.0.2",
```
Comment (Eli250) [inline package.json]: Remove unused package

File: src/app.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ import allRouter from "./routes/index";
 import documentation from "./documentation/index";
 import "dotenv/config";
 import notFound from "./controllers/notFound";
+import "./events/eventHandler/eventHandler";
```
Comment (Eli250) [inline src/app.js:13]: What's the use of this import in this file?

File: tests/expirationProduct.test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable */
```
Comment (Eli250) [inline tests/expirationProduct.test.js]: This file should be named `product.test` and must contain all product-related tests.

File: tests/sample.test.js [hunk 4]
```diff
@@ -1211,7 +1211,23 @@ afterAll(async () => {
   await User.destroy({ where: { email: "testBuyer@gmail.com" } });
 });
 
+describe("Product Expired", ()=>{
```
Comment (Eli250) [inline tests/sample.test.js]: Move these codes to `product.test` file

**Review by Eli250 — CHANGES_REQUESTED** [review]

File: src/app.js (hunk 2 above)
Comment (UwaweTera) [reply src/app.js:13]: It help to run handlers when server is running, when you don't import it nothing happen to handlers

**Review by Eli250 — DISMISSED** [review]

**Review by jimmymut — APPROVED** [review]

**Review by Eli250 — APPROVED** [review]

**Pull request merged**